lock_api/src/mutex.rs

```diff
@@ -681,7 +681,16 @@ unsafe impl<'a, R: RawMutex + 'a, T: ?Sized + 'a> StableAddress for MutexGuard<'
 #[must_use = "if unused the Mutex will immediately unlock"]
 pub struct ArcMutexGuard<R: RawMutex, T: ?Sized> {
     mutex: Arc<Mutex<R, T>>,
-    marker: PhantomData<R::GuardMarker>,
+    marker: PhantomData<*const ()>,
+}
+
+unsafe impl<R: RawMutex + Send + Sync, T: Send + Sync + ?Sized> Send for ArcMutexGuard<R, T> where
```

Comment:
I don't think `T: Sync` is necessary here. Everything else looks fine.

Reply:
Sorry for digging out this old PR, but...

Can you explain why you think that ? The referenced issue was pointing out exactly that.

If I can send a guard to another thread it means that I can access `&T` from another thread, i.e. it must mean that `&T` is `Send` and thus it must mean that `T` is `Sync`.

Reply:
The `ArcMutexGuard` is more like a `&mut T` actually, which also only requires `T: Send` for `&mut T: Send`. This means that if you send the guard, there is still only one thread accessing the `&T` at any one point so `T: !Sync` is still fine.